### DISCUSSION TITLE
Upgrade to latest webpack 5

### DISCUSSION THREAD

**Discussion by karlhorky** [vercel/next.js#26932] [Ideas]
### Describe the feature you'd like to request

`webpack@5.40.0` has added support for the new `node:` prefixes: https://github.com/webpack/webpack/releases/tag/v5.40.0

This will no longer cause an issue with code like:

```js
import fs from 'node:fs'
```

The latest version of webpack at the time of writing is `5.42.0`

cc @ijjk 

### Describe the solution you'd like

Could the compiled version of Node.js be upgraded to this version?

### Describe alternatives you've considered

Continuing to use the prefix-less versions

**Comment by karlhorky** [comment]
@timneutkens oh, this was converted into a discussion? I thought this would be happening sometime pretty soon anyway (kind of like https://github.com/vercel/next.js/pull/26102 by @ijjk ), but I'm not sure what your policy is about upgrading versions of deps (incl. compiled deps)

**Reply by timneutkens** [reply]
All feature requests are converted to discussions (as per the issue template). Unfortunately GitHub does not allow automating that yet so there's a slight delay in converting.

We'll upgrade to the latest version of webpack this week together with some changes for esmodules support.

**Reply by karlhorky** [reply]
> GitHub does not allow automating that yet so there's a slight delay in converting

Ah got it, I understand now. This is a confusing workflow for GitHub users 🤔

> We'll upgrade to the latest version of webpack this week together with some changes for esmodules support.

Great thanks!

**Reply by timneutkens** [reply]
> Ah got it, I understand now. This is a confusing workflow for GitHub users 🤔

We explicitly say this will happen in the issue template. Unfortunately not everyone reads the issue template 😢

**Reply by karlhorky** [reply]
Ah yeah, I did read this (I was aware that feature requests should get converted to discussion)

> Feature requests will be converted to the GitHub Discussions "Ideas" section.

But what I mean is confusing is that it doesn't happen instantly. It made me think that the text above was broken / wrong, and then that this conversion to discussion was a manual action later. Too bad that GitHub doesn't provide an instant way of doing that.

I guess one other thing that led to this assumption was that it was performed by an account associated with a human, instead of one that was identifiably a bot.

**Comment by raitucarp** [comment]
What is status of this issue?

**Reply by timneutkens** [reply]
Next.js runs on 5.61.0 right now: https://github.com/vercel/next.js/blob/canary/packages/next/package.json#L268

**Reply by raitucarp** [reply]
Why do I still getting error cannot load "node:" url scheme?

related issue:
https://github.com/webpack/webpack/issues/13290

**Reply by karlhorky** [reply]
Yeah, the version of webpack in Next.js 12.2.2 still has a problem with the `node:` schema / prefix:

https://github.com/webpack/webpack/issues/13290#issuecomment-1188753208

Workaround (update webpack config in `next.config.mjs`): https://github.com/webpack/webpack/issues/13290#issuecomment-1188760779
